packages/api-providers/src/utils/storage.ts

```diff
@@ -91,3 +90,53 @@ export const keypairStorageFactory = () => {
     },
   });
 };
+
+// Keypair.pubkey -> targetChainId -> noteString
+export type NoteStorage = {
+  encryptedNotes: Record<string, string[]>;
+};
+
+export const noteStorageFactory = (keypair: Keypair) => {
+  // localStorage will have key: Keypair.pubkey, value: Record<targetTypedChainId, EncryptedNote[]>
+  return Storage.newFromCache<NoteStorage>(keypair.pubkey.toString(), {
+    async commit(key: string, data: NoteStorage): Promise<void> {
+      localStorage.setItem(key, JSON.stringify(data));
+    },
+    async fetch(key: string): Promise<NoteStorage> {
+      const storageCached = localStorage.getItem(key);
+
+      if (storageCached) {
+        return {
+          ...JSON.parse(storageCached),
+        };
+      }
+
+      return {
+        encryptedNotes: {},
+      };
+    },
+  });
+};
+
+export const netStorageFactory = () => {
```

Comment:
Should this be `newStorageFactory`?

Reply:
This was moved from the old file `network-storage` into the `storage` file for consistency. It should be `netStorageFactory` or renamed to `networkStorageFactory`.